src/libs/actions/Policy.js

```diff
@@ -447,30 +475,68 @@ function hideWorkspaceAlertMessage(policyID) {
 
 /**
  * @param {String} policyID
- * @param {Object} values
+ * @param {Object} currentCustomUnit
+ * @param {Object} values The new custom unit values
  */
-function setCustomUnit(policyID, values) {
-    DeprecatedAPI.Policy_CustomUnit_Update({
-        policyID: policyID.toString(),
-        customUnit: JSON.stringify(values),
-        lastModified: null,
-    })
-        .then((response) => {
-            if (response.jsonCode !== 200) {
-                throw new Error();
-            }
-
-            updateLocalPolicyValues(policyID, {
-                customUnit: {
-                    id: values.customUnitID,
-                    name: values.name,
-                    value: values.attributes.unit,
+function updateWorkspaceCustomUnit(policyID, currentCustomUnit, values) {
+    const customUnitKey = values.name.toLowerCase();
+    const optimisticData = [
+        {
+            onyxMethod: 'merge',
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                customUnits: {
+                    [customUnitKey]: {
+                        customUnitID: values.customUnitID,
+                        name: values.name,
+                        attributes: values.attributes,
+                        pendingAction: 'update',
+                    },
                 },
-            });
-        }).catch(() => {
-            // Show the user feedback
-            Growl.error(Localize.translateLocal('workspace.editor.genericFailureMessage'), 5000);
-        });
+            },
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: 'merge',
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                customUnits: {
+                    [customUnitKey]: {
+                        pendingAction: null,
+                        errors: null,
+                    },
+                },
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: 'merge',
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                customUnits: {
+                    [customUnitKey]: {
+                        customUnitID: currentCustomUnit.customUnitID,
+                        name: currentCustomUnit.name,
+                        attributes: currentCustomUnit.attributes,
+                        errors: {
+                            [DateUtils.getMicroseconds()]: "Your changes couldn't be saved. The workspace was modified while you were offline, please try again.",
```

Comment:
This is a string that needs to be localized

Reply:
Asked in channel for a translation